posthog/api/test/test_web_vitals.py

```diff
@@ -0,0 +1,160 @@
+from rest_framework import status
+from freezegun import freeze_time
+
+from posthog.test.base import APIBaseTest, ClickhouseTestMixin, _create_event, flush_persons_and_events
+from posthog.models.utils import uuid7
+
+
+class TestWebVitalsAPI(ClickhouseTestMixin, APIBaseTest):
+    def test_web_vitals_no_data(self):
+        response = self.client.get(f"/api/environments/{self.team.pk}/web_vitals/?pathname=/test-path")
+
+        self.assertEqual(response.status_code, status.HTTP_400_BAD_REQUEST)
+        self.assertEqual(response.json(), {"error": "No web vitals data found for this path"})
+
+    def test_web_vitals_missing_pathname(self):
+        response = self.client.get(f"/api/environments/{self.team.pk}/web_vitals/")
+
+        self.assertEqual(response.status_code, status.HTTP_400_BAD_REQUEST)
+        self.assertEqual(
+            response.json(),
+            {
+                "type": "validation_error",
+                "attr": "pathname",
+                "code": "invalid_input",
+                "detail": "This field is required.",
+            },
+        )
+
+    def test_web_vitals_with_data(self):
```

Comment:
Hey @robbie-c, I need some help here. This test is not passing because the API endpoint is not finding any of the events I'm creating here. I don't understand what I'm doing wrong, any clues?

Reply:
Ah, if I tweak the code to `avg` rather than `p90` then it works perfectly 🤯, so there's something else to this

Reply:
Ok, figured it out, just need to fix it at the HogQL level